### PR TITLE
Fix attaching gdb in stress tests

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Should fix
```
++ source attach_gdb.lib
/usr/share/clickhouse-test/ci/stress_tests.lib: line 12: attach_gdb.lib: No such file or directory
```

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit f628cf185cb47d929fe7c30cdeb209009731e70c with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51445/f628cf185cb47d929fe7c30cdeb209009731e70c/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by devcrafter — APPROVED** [review]

**Comment by devcrafter** [comment]
Interesting what happened to performance tests - https://s3.amazonaws.com/clickhouse-test-reports/51445/f628cf185cb47d929fe7c30cdeb209009731e70c/performance_comparison_[3_4]/report.html

**Review by tavplubix — CHANGES_REQUESTED** [review]
https://s3.amazonaws.com/clickhouse-test-reports/51445/62b11a8d9fde642d75d02d5e012acee0b0f20f95/stress_test__debug_/run.log
```
++ source //attach_gdb.lib
/usr/share/clickhouse-test/ci/stress_tests.lib: line 12: //attach_gdb.lib: No such file or directory
```
```
+ attach_gdb_to_clickhouse
/usr/share/clickhouse-test/ci/stress_tests.lib: line 189: attach_gdb_to_clickhouse: command not found
```

**Comment by tavplubix** [comment]
Style Check — Check mypy failed - some mess with docker images

**Comment by Avogar** [comment]
I am confused with stress/upgrade tests failures, not obvious why it failed or I miss something

**Comment by tavplubix** [comment]
I guess that's because we have `set +e` in `run_with_retry` that is called from `attach_gdb_to_clickhouse`, and it's not unset. So, after calling `attach_gdb_to_clickhouse` in `stress/run.sh`, any failed command will terminate the check

**Comment by Avogar** [comment]
But it's unset if command succeded:
https://github.com/ClickHouse/ClickHouse/blob/49c1beb8705399dfb40ae5213ebca70ba31e3852/tests/ci/utils.lib#L14-L16

And if it didn't succed we should have this message I guess:
https://github.com/ClickHouse/ClickHouse/blob/49c1beb8705399dfb40ae5213ebca70ba31e3852/tests/ci/utils.lib#L23

But I dont see it

**Comment by Avogar** [comment]
Yeah, we have `set -e` in run log

**Comment by Avogar** [comment]
Well, removing `run_with_retry` helps

**Comment by tavplubix** [comment]
> But it's unset if command succeded:

Hm, yes, but it appeared that bash is confusing sometimes :confused:  . `set -e` means "_enable_ exit-on-error mode" and `set +e` means "_disable_ exit-on-error mode". So `run_with_retry` enables it before returning, so the `run.sh` script exists on the first failed command after `run_with_retry`

**Comment by Avogar** [comment]
> set -e means "enable exit-on-error mode" and set +e means "disable exit-on-error mode".

Ah, I was confused by this, yes. Thanks :)

File: tests/ci/utils.lib [hunk 1]
```diff
@@ -2,7 +2,12 @@
 
 function run_with_retry()
 {
-    set +e
+    if [[ $- =~e ]]; then
+      set +e
+      set_e=true
+    else
+      set_e=false
+    fi
```
Comment (tavplubix) [inline tests/ci/utils.lib:10]:
```suggestion
    if [[ $- =~e ]]; then
      set_e=true
    else
      set_e=false
    fi
    set +e
```

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
SQLancer (release)  - https://github.com/ClickHouse/ClickHouse/issues/51722

**Pull request merged**